### DISCUSSION TITLE
Disabling iPXE shell by default

### DISCUSSION THREAD

**Discussion by maccadia** [ltsp/ltsp#218] [Q&A]
Currently, LTSP's default iPXE menu gives access to an iPXE shell. Although it's useful for debugging, I think this should be disabled by default for security reasons.

**Comment by alkisg** [comment]
Allowing an iPXE shell is similar to allowing a Grub shell, which AFAIK is the default in Linux distributions.

For that reason, I think both the shell and the config menus should be on by default, and security-sensitive installations can opt to hide them, similarly to what they'd do with Grub.

The following ltsp.conf parameters should allow you to hide them:

```shell
[server]
POST_IPXE_HIDE_CONFIG="sed '/--key c/d' -i /srv/tftp/ltsp/ltsp.ipxe"
POST_IPXE_HIDE_SHELL="sed '/--key s/d' -i /srv/tftp/ltsp/ltsp.ipxe"
```

**Comment by maccadia** [comment]
> Allowing an iPXE shell is similar to allowing a Grub shell

Agreed. This is the default for desktops and servers alike and makes sense if the physical access to those is controlled / secure. Anyone with grub shell can get a root access if there is no disk encryption.

However, I think LTSP is used in contexts where the physical access is not always well controlled (students in my case). Maybe we can add a _security_ section in the official documentation.

**Comment by alkisg** [comment]
> Anyone with grub shell cant get a root access if there is no disk encryption.

I assume you meant "can" there.

> Maybe we can add a security section in the official documentation.

Sounds great; I invite you to start a community wiki page about it and I can pull it or link to it from the official docs.

https://github.com/ltsp/community/wiki/security

**Comment by maccadia** [comment]
Done :-)

**Comment by alkisg** [comment]
Thanks, I linked to it in the https://ltsp.org/docs/ page.

**Comment by enaut** [comment]
@MAccadia - I hope you don't mind my edits (mostly grammar): https://github.com/ltsp/community/wiki/security/_compare/75851c4db9...d8a01f6a00

**Comment by maccadia** [comment]
Not at all. That's what wikis are for ;-)
